hr_utilization_analysis/report/hr_utilization_analysis.py

```diff
@@ -52,18 +52,88 @@ def _compute_entry_ids(self):
-            for employee_id in employee_ids:
-                date = analysis.date_from
-                one_day = timedelta(days=1)
-                while date <= analysis.date_to:
-                    entry_ids.append(
-                        (0, False, {"employee_id": employee_id.id, "date": date})
-                    )
-                    date += one_day
+            dates = []
+            date = analysis.date_from
+            one_day = timedelta(days=1)
+            while date <= analysis.date_to:
+                dates.append(date)
+                date += one_day
+
+            entry_ids = [(5, False, False)] + [
+                (0, False, d) for d in self._get_entry_id_dicts(employee_ids, dates)
+            ]
 
             analysis.entry_ids = entry_ids
 
+    def _get_entry_id_dicts(self, employees, dates):
+        Module = self.env["ir.module.module"]
+        AccountAnalyticLine = self.env["account.analytic.line"]
+
+        project_timesheet_holidays = Module.with_user(SUPERUSER_ID).search(
+            [("name", "=", "project_timesheet_holidays"), ("state", "=", "installed")],
+            limit=1,
+        )
+
+        uom_hour = self.env.ref("uom.product_uom_hour")
+
+        all_line_ids = AccountAnalyticLine.search(
+            [
+                ("project_id", "!=", False),
+                ("employee_id", "in", employees.ids),
+                ("date", "in", dates),
+            ]
+        )
+        entries = []
+        for employee in employees:
+            tz = pytz.timezone(employee.resource_calendar_id.tz)
+            from_datetime = datetime.combine(min(dates), time.min).replace(tzinfo=tz)
+            to_datetime = datetime.combine(max(dates), time.max).replace(tzinfo=tz)
+            work_time = employee.list_work_time_per_day(from_datetime, to_datetime)
+            hours_dict = {w[0]: w[1] for w in work_time}
+            if project_timesheet_holidays:
+                leaves = employee.list_leaves(from_datetime, to_datetime)
+                leaves_dict = {e[0]: e[1] for e in leaves}
+
+            for date in dates:
+                line_ids = all_line_ids.filtered(
+                    lambda l: l.employee_id == employee and l.date == date
+                )
+                base_entry = {"employee_id": employee.id, "date": date}
+
+                capacity = hours_dict.get(date, 0)
+                if project_timesheet_holidays:
+                    capacity -= leaves_dict.get(date, 0)
+                capacity = max(capacity, 0)
+
+                projects = line_ids.mapped("project_id")
+                if not projects:
+                    entry = base_entry.copy()
+                    entry["amount"] = 0.0
+                    entry["line_ids"] = [(4, _id) for _id in line_ids.ids]
+                    entry["capacity"] = entry["difference"] = capacity
+                    entries.append(entry)
+                    continue
+
+                capacity = capacity / len(projects)  # TODO: improve
```

Comment:
That's very questionable approach

Reply:
I don't think I would approve this particular change as of now. Would it be possible to split the PR into 3 separate enhancements?

Reply:
If we could use allocation information per project (from planning), that would be meaningful, but w/o that - ugh, not sure

Reply:
That's true but at least it keeps the report totals correct... to do this correctly I would need to add a `project_id` field to `resource.calendar.attendance` to assign part of someone's schedule to one project and to another, but it's a lot of work.

Do you have another idea?

Reply:
not right now, that's why I propose to split the PR, merge 2 other enhancements and get more feedback project dimension and discuss that more.

Reply:
At least I could make this an overridable function `_get_work_fraction(employee, project)` or something that returns this by default, and other modules can override.

Reply:
`_get_work_fraction` would be good, yet please consider splitting the PR into separate enhancements

Reply:
OK, i'll keep this one about the "project" subject, since we've already done some discussing.

Reply:
thank you!